ProjectEureka/project-eureka-flutter
# PE-48 | Implemented Camera Form to Allow Users to Add Pictures

**What I Did;**
- Used the [Image Picker Package](https://pub.dev/packages/image_picker) to allow users to add images along their "Create a Question Post".
- Used [Photo View Package](https://pub.dev/packages/photo_view) to allow users to open their image and zoom and preview it.
- Used Firebase Storage to upload all images that the user wants and then return the download link of that image.

**How to Test:**
1. I've added a couple of packages, so you will need to run `flutter pub get`.
2. Open the emulator or physical phone and take some pictures to add into the phone's gallery.
3. Create a new question with the button on the bottom of home screen. 
4. The title and description can be what every you want.
5. Then switch to the `Photo Tab` and choose the option to `Take a New Photo`.
6. The screen should update now that an image is added.
7. Take add a couple more pictures but the `Pick Photo from Gallery` option.
8. Then more pictures however you want. 
9. Go back to the `Text` tab and click submit. You then should see a screen saying it's being uploaded.
10. When upload complete, you should see a confirmation of the title, description, and images on the bottom.
11. In you web browser, go to Firebase Storage and you should see the image uploaded there.
